Mustafa Abbas iP

mistyk786 opened the pull request:
# 🚀 Cortana – Your AI Task Manager  

> **"Wake up, Chief. I need you."** – Cortana  

Cortana is your **AI-powered personal assistant** that helps you manage tasks seamlessly. It's:  

- 📝 **Text-based**  
- 🎓 **Easy to learn**  
- ⚡ **FAST – SUPER FAST** to use  

## ✨ Getting Started  
1. 📥 **[Download Cortana](#)** (link)  
2. 🖱️ **Double-click it** to run  
3. ✅ **Add your tasks**  
4. 🤖 **Let Cortana handle them for you!**  

And best of all… **it's FREE!** 🎉  

---

## 📌 Features  
- ✅ **Task management made easy**  
- 📅 **Deadline tracking** _(coming soon)_  
- 🔔 **Smart reminders** _(coming soon)_  

### **Task Capabilities**  
- 🗂️ **Supports 3 types of tasks**:  
  - ✅ `Todo`  
  - 📅 `Deadline`  
  - 📆 `Event`  
- 💾 **Auto-saves task lists**  
- ✏️ **Modify tasks** (mark/unmark/delete)  
- 🔍 **Search tasks using keywords**  

---

## 🖥️ Java Programmers, Here's the Entry Point  
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

mistyk786 committed:
Add Carol.java and delete Duke.java

mistyk786 committed:
Echo message

mistyk786 committed:
Add messages to list

mistyk786 committed:
Added Task file and mark

mistyk786 committed:
ToDos, Events, Deadlines

mistyk786 committed:
Error Handling

mistyk786 committed:
Delete Tasks

mistyk786 committed:
Added SaveTasks and LoadTasks functionality

mistyk786 committed:
Merge branch 'branch-Level-7'

mistyk786 committed:
Add Date and Time Functionality

mistyk786 committed:
Merge branch 'branch-Level-8'

KimHan01 commented on src/main/java/Carol.java:
```diff
@@ -0,0 +1,259 @@
+        out.println(INTRO_MSG);
+        list = loadTasks();
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        boolean running = true;
```
Perhaps its might be better to use "isRunning" instead

KimHan01 commented on src/main/java/Carol.java:
```diff
@@ -0,0 +1,259 @@
+import java.io.*;
```
Imported classes should be listed explicity

KimHan01 commented on src/main/java/Carol.java:
```diff
@@ -0,0 +1,259 @@
+            String eventTimeStart = eventStartString.split(" ")[1];
+            String eventDateEnd = eventEndString.split(" ")[0];
+            String eventTimeEnd = eventEndString.split(" ")[1];
+            Event ev = new Event(eventMessage, LocalDate.parse(eventDateStart), LocalDate.parse(eventDateEnd), LocalTime.parse(eventTimeStart), LocalTime.parse(eventTimeEnd));
```
Line length should be no longer than 120 chars, you could try to bring some arguments into the next line

KimHan01 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,71 @@
+                return null;
+        }
+
+    }
```
A suggestion i have is to allow each Task class (Todo, Deadline, Event) to handle the logic of parsing for their specific task type.

KimHan01 reviewed (commented):
Overall, i think you have several coding standard violations that you may want to address, and you may consider breaking your Main class up into different classes to handle specific logic like a Parser class to handle making sense of user input and a Response/Ui class to handle printing out messages

mistyk786 committed:
Used more OOP

mistyk786 committed:
Merge branch 'branch-MoreOOP'

mistyk786 committed:
Merge branch 'add-gradle-support'

jensenhuangyankai commented on src/main/java/carol/io/EventParser.java:
```diff
@@ -0,0 +1,176 @@
+            DateTimeFormatter.ofPattern("hh:mm a")    // 12-hour format with AM/PM (05:30 PM)
+    };
+
+    public static Task parseTaskFromFile(String line) throws CarolException {
```
Though the purpose of the function is clear, DocStrings are still required for every function

jensenhuangyankai commented on src/main/java/carol/io/Ui.java:
```diff
@@ -0,0 +1,49 @@
+             Hello! I'm Carol, your personal assistant.
+             What can I do for you?
+            """;
+    protected final String END_MSG = " Bye! Hope to hear from you soon!\n";
```
good that you used SCREAMING_SNAKE_CASE to define your constants

jensenhuangyankai commented on src/main/java/carol/io/Ui.java:
```diff
@@ -0,0 +1,49 @@
+        System.out.println(message);
+    }
+
+    public static void line() {
```
This function needs a better name. Consider using printLine instead

jensenhuangyankai commented on src/main/java/carol/io/EventParser.java:
```diff
@@ -0,0 +1,176 @@
+import java.time.format.DateTimeParseException;
+
+public class EventParser {
+    private static final DateTimeFormatter[] DATE_FORMATS = {
```
Consider not using an array for this. Makes code extremely hard to read, as you have to reference what DATE_FORMATS[0] is every time you see it. Maybe do a yyyyMMddFormatter as a name instead.

mistyk786 committed:
Added gradle automation and testing

mistyk786 committed:
Changed class names to match coding standard

mistyk786 committed:
Added JavaDoc comments

mistyk786 committed:
Added Find command

mistyk786 committed:
Merge branch 'A-CodingStandard'

mistyk786 committed:
Merge branch 'A-JavaDoc'

mistyk786 committed:
Merge branch 'Level-9'

nikingoda commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    public void markAsDone() {
+        if (!isDone) {
```
maybe just **isDone = true;** is enough

nikingoda commented on src/main/java/tasks/ToDo.java:
```diff
@@ -0,0 +1,20 @@
+package tasks;
+
+public class ToDo extends Task{
+    private String type;
```
**String type** is not so necessary, or you may change it to final field for safe.

nikingoda commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,37 @@
+import java.time.LocalTime;
+
+public class Event extends Task {
+    private String type;
```
make **String type** to be final could be better

nikingoda commented on src/main/java/io/EventParser.java:
```diff
@@ -0,0 +1,206 @@
+            switch (taskType) {
+            case "T":
+                ToDo todo = new ToDo(description);
+                if (isDone) {
```
Should provide another constructor **Todo(String description, Boolean isDone)**

nikingoda commented on src/main/java/io/EventParser.java:
```diff
@@ -0,0 +1,206 @@
+                LocalTime time = LocalTime.parse(deadlineString, TIME_FORMATS[0]);
+                Deadline deadline = new Deadline(description, date, time);
+                if (isDone) {
+                    deadline.markAsDone();
```
Should provide another constructor **Deadline(String description, LocalDate date, LocalTime time, Boolean isDone)**

nikingoda commented on src/main/java/io/EventParser.java:
```diff
@@ -0,0 +1,206 @@
+                LocalTime eventTimeEnd = LocalTime.parse(eventEndParts[1]);
+
+                Event event = new Event(description, eventDateStart, eventDateEnd, eventTimeStart, eventTimeEnd);
+                if (isDone) {
```
Same with deadline and event.

mistyk786 committed:
Added dependencies

mistyk786 committed:
Added GUI

mistyk786 committed:
Merge branch 'Level-10'

mistyk786 committed:
Fix issues with DateTime and isDone parsing

Loading tasks from storage leads to incorrect isDone value being
returned
Format of LocalDate and LocalTime Strings in Deadline and Event
instances is not standardised

Standardised DateTime values allows for users to read values easier
Correct isDone value ensure user knows when they have marked the task
as done

Let's ensure that tasks are displayed and stored correctly

mistyk786 committed:
Remove duplicate code fragments
Increase code readability

mistyk786 committed:
Merge branch 'branch-A-CodeQuality'

mistyk786 committed:
Add HelpCommand feature

mistyk786 committed:
Fix DateTime parsing to provide correct time

mistyk786 committed:
Merge pull request #3 from mistyk786/branch-helpCommand

Add HelpCommand feature

mistyk786 committed:
Merge branch 'master' into B-Reminders

mistyk786 committed:
Merge pull request #2 from mistyk786/B-Reminders

Fix DateTime parsing to provide correct time

mistyk786 committed:
Push tag

mistyk786 committed:
Create JAR file

mistyk786 committed:
README update

mistyk786 committed:
Fix Ui image

mistyk786 committed:
Add README to docs
